tests/suites/test_suite_shax.function

```diff
@@ -148,3 +149,142 @@ void sha512_selftest()
     TEST_EQUAL(mbedtls_sha512_self_test(1), 0);
 }
 /* END_CASE */
+
```

Comment:
Testing mostly looks good, but we don't have any tests for bad arguments - e.g., passing an invalid hash id, or olen too short.